### PR TITLE
S4 main list page

### PR DESCRIPTION
**Pull request by ucanfil**
Files need to be reviewed:
index.php
scss/index.scss
src/PokeList.php
src/Pokemon.php
pokemonEntryTemplate.phtml

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -10,9 +18,23 @@
...
     <meta charset="UTF-8">
+    <link href="https://fonts.googleapis.com/css?family=Cantarell:400,700" rel="stylesheet">
+    <link rel="stylesheet" href="css/normalize.css">
+    <link rel="stylesheet" href="css/index.css">
     <title>Pokédex</title>
 </head>
 <body>
-
+    <main>
+        <form>
```
Comment (Benson-J) [inline index.php]: Form should not be included until a later story.

File: index.php [hunk 2]
```diff
@@ -10,9 +18,23 @@
...
+            <ul id="scroll">
+                <?php
+                    foreach ($pokeList->getPokemon() as $pokemon) {
+                        require 'pokemonEntryTemplate.phtml';
+                    }
+                ?>
+            </ul>
+            <input type="submit" value="Save" id="save">
```
Comment (Benson-J) [inline index.php]: Save button should not be included until a later story.

File: scss/index.scss [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  float: left;
+}
+
+.lower {
+  height: 70%;
+}
+
+.type-holder {
+  width: 20%;
+  float: left;
```
Comment (Benson-J) [inline scss/index.scss]: Consider removing, as it looks neater without.

File: scss/index.scss [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  background-color: $poke-grey;
+}
+
+.picture {
+  height: 100%;
+  margin: auto;
+  display: block;
+}
+
+#save {
```
Comment (Benson-J) [inline scss/index.scss]: Save button should not be included until a later story.

File: src/PokeList.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+class PokeList {
+
+    private $pokemon;
+
+    /**
+     * Creates a new pokeList using a DBConnection
+     * @param \PDO $connection The particular database object to use for retrieving the list of pokemon.
+     */
+    public function __construct(\PDO $db)
```
Comment (Benson-J) [inline src/PokeList.php:12]: Needs unit testing.

File: src/Pokemon.php [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+    private $id;
+    private $name;
+    private $type_1;
+    private $type_2;
+
+
+    public function __construct()
+    {
+        //capitalises first letter of each string for more a professional appearance
```
Comment (Benson-J) [inline src/Pokemon.php]: Is this comment necessary?  This should probably be in the docblock.

File: src/PokeList.php [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+ * User: academy
+ * Date: 14/11/2018
+ * Time: 11:48
+ */
+
+namespace Pokedex;
+
+class PokeList {
+
+    private $pokemon;
```
Comment (Benson-J) [inline src/PokeList.php]: A bit too general, not clear that this refers to a list of pokemon.

File: src/Pokemon.php [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        $this->name = ucfirst($this->name);
+        $this->type_1 = ucfirst($this->type_1);
+        $this->type_2 = ucfirst($this->type_2);
+    }
+
+    /**
+     * @return int The pokemon's ID.
+     */
+    public function getId()
```
Comment (Benson-J) [inline src/Pokemon.php]: Getters could all be return type hinted.

**Review by Benson-J — CHANGES_REQUESTED** [review]

File: index.php (hunk 1 above)
Comment (ucanfil) [reply index.php]: Solved in 14963fb

File: index.php (hunk 2 above)
Comment (ucanfil) [reply index.php]: Solved in 14963fb

File: scss/index.scss (hunk 3 above)
Comment (ucanfil) [reply scss/index.scss]: Solved in d318df2

File: scss/index.scss (hunk 4 above)
Comment (ucanfil) [reply scss/index.scss]: Solved in d318df2

File: src/Pokemon.php (hunk 6 above)
Comment (ucanfil) [reply src/Pokemon.php]: Solved in b605239

File: src/Pokemon.php (hunk 8 above)
Comment (ucanfil) [reply src/Pokemon.php]: Solved in eb4d541

File: src/PokeList.php (hunk 7 above)
Comment (ucanfil) [reply src/PokeList.php]: Solved in c032493

Comment (ucanfil) [reply src/PokeList.php]: Solved in c032493

File: src/Pokemon.php (hunk 6 above)
Comment (ucanfil) [reply src/Pokemon.php]: Solved in b605239

File: scss/index.scss (hunk 4 above)
Comment (ucanfil) [reply scss/index.scss]: Solved in d318df2

File: scss/index.scss (hunk 3 above)
Comment (ucanfil) [reply scss/index.scss]: Solved in d318df2

File: index.php (hunk 2 above)
Comment (ucanfil) [reply index.php]: Solved in 14963fb

File: index.php (hunk 1 above)
Comment (ucanfil) [reply index.php]: Solved in 14963fb

File: src/Pokemon.php (hunk 8 above)
Comment (ucanfil) [reply src/Pokemon.php]: Solved in eb4d541

File: src/PokeList.php (hunk 5 above)
Comment (ucanfil) [reply src/PokeList.php:12]: Solved in e85a1a8

**Review by Benson-J — APPROVED** [review]

**Review by mporam — APPROVED** [review]

**Pull request merged**